### DISCUSSION TITLE
How does Theory/MemberData attribute determine test execution order?

### DISCUSSION THREAD

**Discussion by nanatech** [xunit/xunit#2170] [Question] [answered]
I've been trying to understand how test execution order is determined when using the Theory tag. I have test data that must be executed sequentially (Not the best practice) inside the MemberData attribute. I'm noticing the tests are executed in the same order each time I run them in Visual studio, but it's not by the order in which I added the data.

**Comment by bradwilson** [comment, marked as answer]
See this issue: https://github.com/xunit/xunit/issues/1301

**Reply by bradwilson** [reply]
Going to mark this as answered. Please let me know if you have any further questions.
